src/vs/workbench/contrib/scm/browser/util.ts

```diff
@@ -37,7 +37,7 @@ export function isSCMResource(element: any): element is ISCMResource {
 	return !!(element as ISCMResource).sourceUri && isSCMResourceGroup((element as ISCMResource).resourceGroup);
 }
 
-const compareActions = (a: IAction, b: IAction) => a.id === b.id;
+const compareActions = (a: IAction, b: IAction) => a.id === b.id && a.enabled === b.enabled;
```

Comment:
Interesting!

Reply:
As discussed, this is necessary in order for the enablement to be refreshed in the repository toolbar.